lib/active_delivery/base.rb

```diff
@@ -73,6 +73,16 @@ def #{line_id}_class
         CODE
       end
 
+      def unregister_line(line_id)
+        removed_line = delivery_lines.delete(line_id)
+
+        return if removed_line.nil?
+
+        [line_id, "#{line_id}_class"].each do |method_name|
+          (class << self; self; end).send(:undef_method, method_name) rescue NameError
```

Comment:
What is the case for NameError here?

Reply:
Should we check if method exists before undefining it? 

Was just worried it may be the case the method was removed by some other means and then this unregester_line would raise an error when really it should just silently go about its business if the method is not there already.

Reply:
And I am not a meta-whiz when it comes to Ruby so maybe there is a better way to re-open the singleton class and use the variable than `(class << self; self; end).send` ... but that's what I could sort out to work.

Reply:
Ah — rubocop.  Let me know if you want to drop the rescue block or use a `respond_to?` instead and I will fix the linting.

Reply:
> maybe there is a better way to re-open the singleton class 

yeah, we can access it via `singleton_class`: i.e. `singleton_class.undef_method :foo`.

> it may be the case the method was removed by some other means

I think, it's better to fail here to let users know that they're doing something strange (usually, removing not your own methods is a bad idea).

`return if removed_line.nil?` is enough here.

Reply:
Ah that is so much better!  Forgot about `singleton_class`. Sigh. 🤷 

Rebased and hopefully tests are passing now.